pkg/controller/v1alpha1/utils/utils.go

```diff
@@ -93,28 +97,30 @@ func SetAutoScalingAnnotations(ctx context.Context,
 		)
 		annotations[autoscaling.InitialScaleAnnotationKey] = annotations[autoscaling.MinScaleAnnotationKey]
 	} else {
-		log.Info(
-			fmt.Sprintf(
-				"The %s annotation is explicitly set. kserve will override the default initial scale value for knative revisions with the value of this annotation",
-				autoscaling.InitialScaleAnnotationKey,
-			),
-			"initial-scale", annotations[autoscaling.InitialScaleAnnotationKey],
-		)
-	}
-	initialScale, err := strconv.Atoi(annotations[autoscaling.InitialScaleAnnotationKey])
-	if err != nil {
-		return errors.Wrapf(
-			err,
-			"failed to convert %s annotation value of %s to an integer",
-			autoscaling.InitialScaleAnnotationKey,
-			annotations[autoscaling.InitialScaleAnnotationKey],
-		)
-	}
-
-	// Retrieve the allow-zero-initial-scale value from the knative autoscaler configuration.
-	allowZeroInitialScale, err := CheckZeroInitialScaleAllowed(ctx, clientset)
-	if err != nil {
-		return errors.Wrapf(err, "failed to retrieve the knative autoscaler configuration")
+		// If the annotation is explicitly set, validate that it is set to an integer value.
```

Comment:
@brettmthompson See https://github.com/kserve/kserve/issues/4486, on line 98 can we not default to MinScale as that's the changing the default Knative behavior.